diff -uprN debian/control debian_mod/control
--- debian/control	2018-02-08 03:52:41.000000000 +0900
+++ debian_mod/control	2020-01-31 10:15:57.380574014 +0900
@@ -11,16 +11,16 @@ Build-Depends: debhelper (>= 9),
   omniorb-idl (>= 4.2.2),
   omniidl (>= 4.2.2),
   autotools-dev
-Build-Conflicts: omniidl-python
+Build-Conflicts: omniidl-python3
 Standards-Version: 4.1.3
 Vcs-Svn: svn://anonscm.debian.org/pkg-corba/trunk/python-omniorb
 Vcs-Browser: http://anonscm.debian.org/viewvc/pkg-corba/trunk/python-omniorb
 Homepage: http://omniorb.sourceforge.net
 
-Package: python-omniorb
+Package: python3-omniorb
 Architecture: any
-Depends: ${python:Depends}, ${shlibs:Depends}, ${misc:Depends}
-Recommends: python-omniorb-omg
+Depends: ${python3:Depends}, ${shlibs:Depends}, ${misc:Depends}
+Recommends: python3-omniorb-omg
 Description: Python bindings for omniORB
  omniORB4 is a freely available Common Object Request Broker
  Architecture (CORBA) 2.6 compliant object request broker (ORB)
@@ -31,21 +31,21 @@ Description: Python bindings for omniORB
  This is the Debian package of omniORBpy, the Python bindings to the
  omniORB libraries.
 
-Package: python-omniorb-dbg
-Priority: optional
-Architecture: any
-Depends: python-omniorb (= ${binary:Version}), ${python:Depends}, ${shlibs:Depends}, ${misc:Depends}
-Recommends: python-dbg
-Section: debug
-Description: Python bindings for omniORB
- omniORB4 is a freely available Common Object Request Broker
- Architecture (CORBA) 2.6 compliant object request broker (ORB)
- implementation. It is based on the IIOP communications
- protocol and should be interoperable with any other
- CORBA 2.6 compliant ORB.
- .
- This package contains the debug symbols of python-omniorb as well as
- modules for use with python-dbg.
+#Package: python3-omniorb-dbg
+#Priority: optional
+#Architecture: any
+#Depends: python3-omniorb (= ${binary:Version}), ${python3:Depends}, ${shlibs:Depends}, ${misc:Depends}
+#Recommends: python-dbg
+#Section: debug
+#Description: Python bindings for omniORB
+# omniORB4 is a freely available Common Object Request Broker
+# Architecture (CORBA) 2.6 compliant object request broker (ORB)
+# implementation. It is based on the IIOP communications
+# protocol and should be interoperable with any other
+# CORBA 2.6 compliant ORB.
+# .
+# This package contains the debug symbols of python-omniorb as well as
+# modules for use with python-dbg.
 
 Package: python-omniorb-doc
 Architecture: all
@@ -62,9 +62,9 @@ Description: omniORBpy documentation
  bindings to omniORB.  The bindings themselves can be found in the
  python-omniorb package.
  
-Package: python-omniorb-omg
+Package: python3-omniorb-omg
 Architecture: all
-Depends: python-omniorb (>= ${binary:Version}), ${python:Depends}, ${misc:Depends}
+Depends: python3-omniorb (>= ${binary:Version}), ${python3:Depends}, ${misc:Depends}
 Conflicts: python-pyorbit-omg
 Description: CORBA OMG standard files for python-omniorb
  omniORB4 is a freely available Common Object Request Broker
@@ -78,10 +78,10 @@ Description: CORBA OMG standard files fo
  python-pyorbit-omg since only one package can provide the default
  CORBA bindings.
 
-Package: omniidl-python
+Package: omniidl-python3
 Architecture: all
-Depends: omniidl, ${python:Depends}, ${misc:Depends}
-Recommends: python-omniorb
+Depends: omniidl, ${python3:Depends}, ${misc:Depends}
+Recommends: python3-omniorb
 Description: omniidl backend to compile Python stubs from IDL files
  omniORB4 is a freely available Common Object Request Broker
  Architecture (CORBA) 2.6 compliant object request broker (ORB)
diff -uprN debian/omniidl-python3.install debian_mod/omniidl-python3.install
--- debian/omniidl-python3.install	1970-01-01 09:00:00.000000000 +0900
+++ debian_mod/omniidl-python3.install	2020-01-31 10:15:57.380574014 +0900
@@ -0,0 +1 @@
+usr/lib/python*/*-packages/omniidl_be/python.py usr/lib/omniidl/omniidl_be/
diff -uprN debian/python3-omniorb-omg.install debian_mod/python3-omniorb-omg.install
--- debian/python3-omniorb-omg.install	1970-01-01 09:00:00.000000000 +0900
+++ debian_mod/python3-omniorb-omg.install	2020-01-31 10:15:57.380574014 +0900
@@ -0,0 +1,2 @@
+usr/lib/python*/*-packages/*.py
+usr/lib/python*/*-packages/CosNaming*
diff -uprN debian/python3-omniorb.install debian_mod/python3-omniorb.install
--- debian/python3-omniorb.install	1970-01-01 09:00:00.000000000 +0900
+++ debian_mod/python3-omniorb.install	2020-01-31 10:15:57.380574014 +0900
@@ -0,0 +1,3 @@
+usr/lib/python*/*-packages/omniORB.pth
+usr/lib/python*/*-packages/_omni*-??m-*.so*
+usr/lib/python*/*-packages/omniORB/
diff -uprN debian/rules debian_mod/rules
--- debian/rules	2018-02-08 03:52:41.000000000 +0900
+++ debian_mod/rules	2020-01-31 10:15:57.380574014 +0900
@@ -23,7 +23,7 @@ ifeq (,$(findstring noopt,$(DEB_BUILD_OP
   CXX += -O2
 endif
 
-PYVERS := $(shell pyversions -vr debian/control)
+PYVERS := $(shell py3versions -vr debian/control)
 
 CONFIGURE = \
 	CC="$(CC)" CXX="$(CXX)" ../configure $(confflags) \
@@ -86,9 +86,9 @@ install: build
 	done
 	find . -name "*.pyc" -exec rm {} \;
 	find debian/tmp* -name '_omni*.so*'
-	for i in $$(find debian/tmp-dbg -name '_omni*._d.so*'); do \
+	for i in $$(find debian/tmp-dbg -name '_omni*dm-*.so*'); do \
 	  echo "Renaming $$i ..."; \
-	  dst=$$(echo $$i | sed 's,tmp-dbg,tmp,;s/\._d\.so/_d.so/'); \
+	  dst=$$(echo $$i | sed 's,tmp-dbg,tmp,;'); \
 	  mv $$i $$dst; \
 	done
 	find debian/tmp -name '_omni*.so*'
@@ -102,7 +102,7 @@ binary-indep: build install
 	dh_installdocs -i
 	dh_installexamples -i
 	dh_installchangelogs update.log -i
-	dh_python2 -i
+	dh_python3 -i
 	dh_link -i
 	dh_compress -i -X.pdf
 	dh_fixperms -i
@@ -117,8 +117,10 @@ binary-arch: build install
 	dh_testroot -a
 	dh_installdocs -a
 	dh_installchangelogs update.log -a
-	dh_python2 -a
-	dh_strip -ppython-omniorb --dbg-package=python-omniorb-dbg
+	#dh_python3 -a
+	#dh_strip -ppython3-omniorb --dbg-package=python3-omniorb-dbg
+	mv debian/python3-omniorb/usr/lib/python$(PYVERS)/site-packages debian/python3-omniorb/usr/lib/python$(PYVERS)/dist-packages
+	mv debian/python3-omniorb/usr/lib/python$(PYVERS) debian/python3-omniorb/usr/lib/python3
 	dh_link -a
 	dh_compress -a
 	dh_fixperms -a
